frontend/src/ReviewNFT.js

```diff
@@ -0,0 +1,69 @@
+import './styles/style.css';
+import React, { useEffect, useState } from "react";
+import { useNavigate, useSearchParams } from "react-router-dom";
+import { authenticateAction } from "./comps/authenticate"
+import axios from "axios";
+
+function ReviewNFT(){
+
+    const [nft, setNFT] = useState({});
+    const [searchParams] = useSearchParams();
+    const navigate = useNavigate();
+    const [NFTReview, setNFTReview] = useState();
+
+    useEffect(() => {
+        const query = searchParams.get('id')
+        axios.get("/api/nft/listing", {
+            params:{
+               id: query 
+            }
+        }).then((res) => {
+            setNFT(res.data)
+        });
+    }, []);
+
+    async function review(){
+        const [publicAddress, signature] = await authenticateAction(navigate);
+        const response = await axios.put("/api/nft/listing", {
+            listingID: nft._id, 
+            nftReview: NFTReview
+        
+            }, {
+                params:
+                {        
+                publicAddress: publicAddress,
+                signature: signature
+                }
+        })
+    
+    }
```

Comment:
api/nft/listing doesn't work like this. Additionally, you're not editing an NFT rental listing.